axum/src/routing/method_filter.rs

```diff
@@ -21,3 +23,40 @@ bitflags! {
         const TRACE =   0b100000000;
     }
 }
+
+#[derive(Debug)]
+pub struct NoMatchingMethodFilter {
+    method: http::Method,
+}
+
+impl NoMatchingMethodFilter {
+    pub fn method(&self) -> &http::Method {
```

Comment:
Also need docs here.

Reply:
thank you for all the help ! Just pushed but struggled hard to get a decent doc so I didn't do _Examples_ and I'm not sure if I did something wrong on the intra-doc links or its just my IDE